source/beamline/BioXAS/BioXASSSRLMonochromator.h

```diff
@@ -0,0 +1,104 @@
+#ifndef BIOXASSSRLMONOCHROMATOR_H
+#define BIOXASSSRLMONOCHROMATOR_H
+
+#include <QObject>
+
+#include <actions3/AMAction3.h>
+
+#define BRAGG_MOTOR_CRYSTAL_CHANGE_POSITION 55.0
+
+
+class BioXASSSRLMonochromator : public QObject
+{
+	Q_OBJECT
+
+public:
+	/// Enumerates the different region states.
+	class Region { public: enum State { None = 0, A, B }; };
```

Comment:
@helfrij 
This is kinda cool. I've always had a problem with enum names in classes, because I would do something like
enum RegionState{ None, A, B} but then try to call it from another class and the name BioXASSSRLMonochromator::A wasn't that descriptive so I would change to enum RegionState{ RegionNone, RegionA, RegionB} which served the purpose, but seemed overly wordy.
With your way you can call BioXASSSRLMonochromator::Region::A from another class and be both descriptive and terse at the same time. Nice idea.